### PR TITLE
Fixes for potrf and potrs API incompatibility

### PR DESCRIPTION
**Pull request by KulikovNikita**
Changes in API happened and adapters should be modified respectively

### CODE REVIEW

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h [hunk 1]
```diff
@@ -50,15 +50,21 @@ struct MKLPotrf
...
-        cl::sycl::buffer<int64_t, 1> info(cl::sycl::range<1>(1));
 
-        fpk::lapack::potrf(_queue, uplomkl, n, a_sycl_buff, lda, info);
+        {
+            using namespace daal::services;
+            const std::int64_t minimal_scratchpad_size = fpk::lapack::potrf_scratchpad_size<algorithmFPType>(_queue, uplomkl, n, lda);
+            if (scratchpad.get_count() < minimal_scratchpad_size) return Status(ErrorID::ErrorMemoryAllocationFailed);
+        }
+
+        //Arguments order: queue, uplo, n, &a, lda, &scratchpad, scratchpad_size
```
Comment (rlnx) [inline cpp/daal/include/sycl/internal/math/mkl_lapack.h:66]: The comment is useless, remove it

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h [hunk 2]
```diff
@@ -50,15 +50,21 @@ struct MKLPotrf
...
         services::Status status;
-
         const fpk::uplo uplomkl                          = uplo == math::UpLo::Upper ? fpk::uplo::upper : fpk::uplo::lower;
         cl::sycl::buffer<algorithmFPType, 1> a_sycl_buff = a.toSycl();
-        cl::sycl::buffer<int64_t, 1> info(cl::sycl::range<1>(1));
 
-        fpk::lapack::potrf(_queue, uplomkl, n, a_sycl_buff, lda, info);
+        {
+            using namespace daal::services;
+            const std::int64_t minimal_scratchpad_size = fpk::lapack::potrf_scratchpad_size<algorithmFPType>(_queue, uplomkl, n, lda);
```
Comment (rlnx) [inline cpp/daal/include/sycl/internal/math/mkl_lapack.h:62]: Code style: `minimal_scratchpad_size` -> `minimalScratchpadSize `

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h [hunk 3]
```diff
@@ -67,6 +73,20 @@ struct MKLPotrf
         return status;
     }
 
+    services::Status operator()(const math::UpLo uplo, const size_t n, services::Buffer<algorithmFPType> & a, const size_t lda,
+                                const std::int64_t scratchpad_size)
```
Comment (rlnx) [inline cpp/daal/include/sycl/internal/math/mkl_lapack.h:77]: Code style: `scratchpad_size` -> `scratchpadSize`

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h [hunk 4]
```diff
@@ -99,6 +123,22 @@ struct MKLPotrs
...
+    {
+        cl::sycl::buffer<algorithmFPType, 1> scratchpad_buffer { cl::sycl::range<1>(scratchpad_size) };
+        return this->operator()(uplo, n, ny, a, lda, b, ldb, scratchpad_buffer);
+    }
+
+    services::Status operator()(const math::UpLo uplo, const size_t n, const size_t ny, services::Buffer<algorithmFPType> & a, const size_t lda,
+                                services::Buffer<algorithmFPType> & b, const size_t ldb)
+    {
+        const fpk::uplo uplomkl = uplo == math::UpLo::Upper ? fpk::uplo::upper : fpk::uplo::lower;
+        //Arguments Order: queue, uplo, n, ny, lda, ldb
```
Comment (rlnx) [inline cpp/daal/include/sycl/internal/math/mkl_lapack.h:137]: Comment is useless

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h [hunk 5]
```diff
@@ -67,6 +73,20 @@ struct MKLPotrf
...
+    {
+        cl::sycl::buffer<algorithmFPType, 1> scratchpad_buffer { cl::sycl::range<1>(scratchpad_size) };
+        return this->operator()(uplo, n, a, lda, scratchpad_buffer);
+    }
+
+    services::Status operator()(const math::UpLo uplo, const size_t n, services::Buffer<algorithmFPType> & a, const size_t lda)
+    {
+        const fpk::uplo uplomkl                    = uplo == math::UpLo::Upper ? fpk::uplo::upper : fpk::uplo::lower;
+        const std::int64_t minimal_scratchpad_size = fpk::lapack::potrf_scratchpad_size<algorithmFPType>(_queue, uplomkl, n, lda);
+        return this->operator()(uplo, n, a, lda, minimal_scratchpad_size);
```
Comment (rlnx) [inline cpp/daal/include/sycl/internal/math/mkl_lapack.h:87]: Is `potrf_scratchpad_size` expensive operation? The problem is one of the operator overloadings compares buffers size with the value returned by `potrf_scratchpad_size`. In that function, we use `potrf_scratchpad_size` to determine required buffer size, so check for error in the overloading just adds overhead.

If `potrf_scratchpad_size` is cheap, ignore this comment.

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 4 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h:137]: I suppose that it is not harmful too.

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 5 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h:87]: It seems to be very cheap.

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 1 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h]: Fixed

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 3 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h]: Fixed

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 4 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h]: Fixed

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 2 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h]: Fixed

File: cpp/daal/include/sycl/internal/math/mkl_lapack.h (hunk 5 above)
Comment (KulikovNikita) [reply cpp/daal/include/sycl/internal/math/mkl_lapack.h]: I've checked an assembly at it's pretty compact and computationally cheap function

**Pull request merged**